### PR TITLE
Use intermediate device local gradient buffers for backwards rasterization

### PR DESCRIPTION
**Pull request by matthewdcong**
Currently, backwards rasterization for mGPU suffers degraded performance due to atomics and page contention. Fix this by switching to device local buffers for gradient accumulation followed by a NCCL `ReduceScatter` operation to compute the final accumulated output across GPUs in managed memory.

Even on the small MipNeRF 360 example training for 10 epochs, this reduces the runtime by 35%.

### CODE REVIEW

File: src/fvdb/detail/ops/gsplat/RasterizeScreenSpaceGaussiansBackward.cu [hunk 1]
```diff
@@ -1270,6 +1334,51 @@ callRasterizeBackwardPrivateUse1(
...
+        const size_t conicsBytes    = conics.numel() * conics.element_size();
+        const size_t featuresBytes  = features.numel() * features.element_size();
+        const size_t opacitiesBytes = opacities.numel() * opacities.element_size();
+
+        ScalarType *&outDLossDMeans2DLocalPtr = outDLossDMeans2DLocalPtrs[deviceId];
+        C10_CUDA_CHECK(cudaMallocAsync(&outDLossDMeans2DLocalPtr, means2dBytes, stream));
+        C10_CUDA_CHECK(cudaMemsetAsync(outDLossDMeans2DLocalPtr, 0, means2dBytes, stream));
+        torch::Tensor &outDLossDMeans2DLocal = outDLossDMeans2DLocals[deviceId];
+        outDLossDMeans2DLocal =
+            at::from_blob(outDLossDMeans2DLocalPtr, means2d.sizes(), localTensorOptions);
```
Comment (swahtz) [inline src/fvdb/detail/ops/gsplat/RasterizeScreenSpaceGaussiansBackward.cu:1350]: **❓ question:** ‏I'm wondering, why do we need to allocate the tensors like this, can't we do something like:

```
c10::cuda::CUDAGuard deviceGuard(deviceId);
c10::cuda::CUDAStreamGuard streamGuard(stream);
auto outDLossDMeans2DLocal = at::zeros(means2d.sizes(), localTensorOptions);
// more tensors
```

And avoid the state where a potential error might leave us in where we don't hit the `cudaFreeAsync` block below?

**Review by swahtz — APPROVED** [review]
lgtm, just one question but don't want to block

File: src/fvdb/detail/ops/gsplat/RasterizeScreenSpaceGaussiansBackward.cu (hunk 1 above)
Comment (matthewdcong) [reply src/fvdb/detail/ops/gsplat/RasterizeScreenSpaceGaussiansBackward.cu:1350]: See https://github.com/openvdb/fvdb-core/pull/703#discussion_r3653757064

Comment (swahtz) [reply src/fvdb/detail/ops/gsplat/RasterizeScreenSpaceGaussiansBackward.cu:1350]: Ah I see, ok, too bad.
